### ISSUE TITLE
[RFC] Support workflow with forks

### ISSUE THREAD

**Issue by ChShersh** [kowainik/hit-on#66]
`hit` is very convenient when working with repos you own. But sometimes I still need to work with forks. I wonder, whether it's possible to make `hit` work with forks without sacrificing current UI.

My typical workflow with forks include:

1. Fork repo on GitHub.
2. Clone my fork (already can be achieved by `hit clone` command).
3. Do my work (all `hit` command work here)
4. Show all remotes (can be `hit remotes` command)
5. Add remote of the original repo (can be `hit add-remote <owner>/<repo>`)
6. Rebase on the branch from the remote repo (probably can patch `hit fresh` command so it understands `hit fresh <remote>/<repo>`).
7. Push and open PR.

What do you think?

**Comment by krisis** [comment]
@chshersh It would be awesome to have feature in `hit`.
I have a question regarding how listing of issues would work in the presence of forks. Usually the upstream repository is where the issues are maintained, so `hit issue` should be listing issues from the upstream repository, right?

**Comment by chshersh** [comment]
@krisis We haven't designed this feature entirely yet. But regarding your question: forks also can have issues. See this repo, for example:

* https://github.com/phadej/postgresql-simple

Also, we can't know locally whether this package is a fork or not. So I would say that by default all commands should work with the repo itself.

**Comment by vrom911** [comment]
@chshersh, actually you can get this info: https://github.com/phadej/github/blob/1eb7257055b693d17b68e3f454c574accc618cb0/src/GitHub/Data/Repos.hs#L39

Also, working with forks as with the compelely independent libraries is not a different use case. Another thing is when you fork but want to work with the original repository (just in my understanding)...

**Comment by chshersh** [comment]
@krisis We haven't designed this feature entirely yet. But I tend to agree that listing of all issues should be from the upstream repository by default. And since we're using `github` library, we can get the information about whether the library is a fork or not:

* https://github.com/phadej/github/blob/1eb7257055b693d17b68e3f454c574accc618cb0/src/GitHub/Data/Repos.hs#L39

Note, that even if forked projects can have their own issues as well (like the following library):

* https://github.com/phadej/postgresql-simple

But I guess a more typical workflow with forked libraries is when you're not a maintainer, but rather a contributor. And you want to list all issues from the original repo, not from the fork.

**Comment by krisis** [comment]
@chshersh, I agree with the points you make above. I hadn't thought through the different combinations of forks vs owned repositories and maintainer vs contributor when I made the comment earlier.

This is how I imagine the workflow for a contributor working on forked project to be,
  1. Fork your favorite project on github.com
  e.g, `kowainik/hit-on` to `krisis/hit-on`

  2. Clone the repo 
  ```shell
  hit clone hit-on
  ```
  This should,
  - Clone the repo from your fork (if available), else fail the command
  - Set origin and upstream accordingly like e.g, upstream would be `kowainik/hit-on` and origin would be `krisis/hit-on`
  
  3. List the issues (upstream) that you can pick and work on
  ```shell
  hit issue 
  ```
  
  4. If the contributor wishes to track personal development milestones using issues in their fork
  We could make it possible using a boolean flag like,
  ```shell
  hit issue --local
  ```

Is this close to how you are thinking about this?
